### PR TITLE
[Side panel] Create new observation when #o=:new

### LINKED ISSUE #10: [Feature panel] Add observation
User selects a map feature and clicks the "Add" button in the side panel to start filling out a new empty form.

@hintzemichael Could you kindly paste mocks for this (possibly w/redlines) into this issue?

### PR DESCRIPTION
**Pull request by gyarill**
Fixes #10.

### CODE REVIEW

File: web-ng/src/app/services/observation/observation.service.ts [hunk 1]
```diff
@@ -41,35 +52,71 @@ export class ObservationService {
...
 
+  createNewObservation(
+    user: User | null | undefined,
+    project: Project,
+    feature: Feature
+  ): Observation | LoadingState {
+    if (user === null || user === undefined) {
+      return LoadingState.LOADING_FAILED;
+    }
+    const form: Form | undefined = project
```
Comment (gino-m) [inline web-ng/src/app/services/observation/observation.service.ts:98]: Can the type here be inferred instead of declaring it explicitly?

Comment (gyarill) [reply web-ng/src/app/services/observation/observation.service.ts]: ok.. why do you prefer inferred types? aren't declared types more readable and safer?

File: web-ng/src/app/services/observation/observation.service.ts [hunk 2]
```diff
@@ -41,35 +52,71 @@ export class ObservationService {
...
+      .getLayer(feature.layerId)!
+      .forms?.first();
+    if (form === undefined) {
+      return LoadingState.LOADING_FAILED;
```
[the remark this answers is not in the document]
Comment (gyarill) [reply web-ng/src/app/services/observation/observation.service.ts]: I deleted LOADING_FAILED entirely

File: web-ng/src/app/services/observation/observation.service.ts (hunk 1 above)
Comment (gino-m) [reply web-ng/src/app/services/observation/observation.service.ts]: Mainly for consistency; if we do it here we should do it everywhere, but that would make the code less readable overall. Wdyt?

File: web-ng/src/app/services/observation/observation.service.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gino-m) [reply web-ng/src/app/services/observation/observation.service.ts]: Sgtm,

**Comment by gino-m** [comment]
Note that there are a few pending lint errors:
```

/workspace/web-ng/src/app/services/observation/observation.service.ts
   38:10  warning  'Form' is defined but never used                                                                                                                                                                                       @typescript-eslint/no-unused-vars
   95:17  error    Delete `··`                                                                                                                                                                                                            prettier/prettier
   95:17  error    Trailing spaces not allowed                                                                                                                                                                                            no-trailing-spaces
   96:19  warning  Strings must use singlequote                                                                                                                                                                                           quotes
   96:63  error    Insert `;`                                                                                                                                                                                                             prettier/prettier
  100:37  error    Insert `·`                                                                                                                                                                                                             prettier/prettier
  102:19  error    Replace ``Layer·${feature.layerId}·does·not·contain·any·forms·->·Can·not·create·new·observation.`)` with `⏎········`Layer·${feature.layerId}·does·not·contain·any·forms·->·Can·not·create·new·observation.`⏎······);`  prettier/prettier
```

Some can be fixed with `npm run fix`.

File: web-ng/src/app/services/observation/observation.service.ts (hunk 1 above)
Comment (gyarill) [reply web-ng/src/app/services/observation/observation.service.ts]: i am fine either way. I changed it already so there is no type here

**Pull request merged**